src/drivers/qnx/lv_qnx.c

```diff
@@ -0,0 +1,539 @@
+/**
+ * @file lv_qnx.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "lv_qnx.h"
+#if LV_USE_QNX
+#include <stdbool.h>
+#include "../../core/lv_refr.h"
+#include "../../stdlib/lv_string.h"
+#include "../../core/lv_global.h"
+#include "../../display/lv_display_private.h"
+#include "../../lv_init.h"
+#include <stdlib.h>
+#include <time.h>
+#include <screen/screen.h>
+#include <sys/keycodes.h>
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+typedef struct {
+    screen_window_t     window;
+    screen_buffer_t     buffers[LV_QNX_BUF_COUNT];
+    int                 bufidx;
+    bool                managed;
+    lv_indev_t     *    pointer;
+    lv_indev_t     *    keyboard;
+} lv_qnx_window_t;
+
+typedef struct {
+    int                 pos[2];
+    int                 buttons;
+} lv_qnx_pointer_t;
+
+typedef struct {
+    int                 key;
+    int                 flags;
+} lv_qnx_keyboard_t;
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+static uint32_t get_ticks(void);
+static void flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * color_p);
+static bool window_create(lv_display_t * disp);
+static bool init_display_from_window(lv_display_t * disp);
+static void get_pointer(lv_indev_t * indev, lv_indev_data_t * data);
+static void get_key(lv_indev_t * indev, lv_indev_data_t * data);
+static bool handle_pointer_event(lv_display_t * disp, screen_event_t event);
+static bool handle_keyboard_event(lv_display_t * disp, screen_event_t event);
+static void release_disp_cb(lv_event_t * e);
+static void refresh_cb(lv_timer_t * timer);
+
+/***********************
+ *   GLOBAL PROTOTYPES
+ ***********************/
+
+static screen_context_t context;
+static bool inited = false;
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+lv_display_t * lv_qnx_window_create(int32_t hor_res, int32_t ver_res)
+{
+    if(!inited) {
```

Comment:
We can move the `inited` variable declaration here since it is only accessed in this function.